### DISCUSSION TITLE
Unable to hear audio in downloaded videos in QuickTime Player

### DISCUSSION THREAD

**Discussion by Labhansh-Sharma** [section83/MacYTDL#55] [Q&A] [answered]
I have tried multiple settings combinations i.e remux format, audio format etc but I'm still unable to hear audio. Even though audio codec is set to "aac", downloaded video still has opus codec. 
Downloaded videos works well without any problems in infuse, vlc etc but not in QuickTime
<img width="512" alt="Screenshot 2022-05-31 at 11 28 30 PM" src="https://user-images.githubusercontent.com/72038652/171254028-4061df75-4d00-48d3-9790-c7588eba4078.png">

<img width="372" alt="Screenshot 2022-05-31 at 11 25 17 PM" src="https://user-images.githubusercontent.com/72038652/171254058-da567c83-c896-4a35-b1ce-25548484a308.png">

<img width="592" alt="Screenshot 2022-05-31 at 11 27 10 PM" src="https://user-images.githubusercontent.com/72038652/171254080-4313e844-a02e-4de2-a166-7eb81ddfcdd5.png">

Please let me know that what should be the correct settings configuration so that I can open videos in QuickTime without any problems.

**Comment by section83** [comment]
Labhansh-Sharma, thank you for making contact.  Sorry you are having this problem.  The "Audio format:" setting is only used when "Audio only" is ticked. So, it won't help with your problem.

To play downloaded YouTube videos and the like in Quicktime try this:

- Delete the previously downloaded video (that one which doesn't work).
- Go to “Settings” (command-s).
- Click on “Use custom settings”.
- Copy the following text and paste into the text box which is beside “Use custom settings”

        --recode-video "mp4" --ppa "VideoConvertor:-vcodec libx264 -acodec aac"
- Make sure that the double quotes are NOT “smart” quotes.
- Make sure not to copy the line break after “aac"”.
- I recommend you also turn on “Verbose logging” and “Add timestamps to log”.
- Set "Remux format:" to "No remux".
- Make sure "Audio only" is NOT ticked.
- Click on "Save Settings".
- Try to download again.

Your settings will look like this:

<img width="480" alt="Settings" src="https://user-images.githubusercontent.com/7471595/171302453-ccc05482-966d-4ad7-8841-e8a3a73ccbdf.png">


Post here again if you have any problems or questions.

Cheers.

**Reply by Labhansh-Sharma** [reply]
Thank you so much Sir, I can finally hear audio in QuickTime itself. 

I've 1 problem though - If I'm downloading a 10 minute long video, CPU usage can be more than 80%(This wasn't the case before while downloading videos with wrong settings😅)

**Comment by section83** [comment, marked as answer]
Very glad it has worked.

> If I'm downloading a 10 minute long video, CPU usage can be more than 80%

You don't say whether CPU usage is 80% while downloading or while FFmpeg is processing the file. I suspect it's the latter.  If using the settings I suggested, after the download is completed, FFmpeg is started to re-encode all the video and audio data in the file.  Re-encoding that data is CPU intensive. The amount of CPU used depends mostly on the complexity of the data (e.g. video resolution, frame rate etc.). But, on a slow Mac it can take a long time to re-encode a large file.

I believe the only way to reduce FFmpeg's call on the CPU is to reduce it's priority in macOS's process management – I've never done it. It should not impede other work though.
